### PR TITLE
[1LP][RFR] Wait for embedded ansible only when enabling it

### PR DESCRIPTION
**Pull request by jkrocil**
Fix for 16+ tests

### CODE REVIEW

File: cfme/utils/appliance/__init__.py [hunk 1]
```diff
@@ -1726,13 +1726,18 @@ def server_roles(self, roles):
         if self.server_roles == roles:
             self.log.debug(' Roles already match, returning...')
             return
-        ansible = roles.get('embedded_ansible', False)
+        ansible_old = self.server_roles('embedded_ansible', False)
+        ansible_new = roles.get('embedded_ansible', False)
+        enabling_ansible = False
+        if ansible_old is False and ansible_new is True:
```
Comment (mshriver) [inline cfme/utils/appliance/__init__.py]: consider just setting `enabling_ansible` to the result of the boolean statement.

```python
enabling = (old is False and new is True)

**Comment by jkrocil** [comment]
@quarckster

File: cfme/utils/appliance/__init__.py [hunk 2]
```diff
@@ -1726,13 +1726,18 @@ def server_roles(self, roles):
...
+            enabling_ansible = True
+
         yaml = self.get_yaml_config()
         yaml['server']['role'] = ','.join([role for role, boolean in roles.iteritems() if boolean])
         self.set_yaml_config(yaml)
-        timeout = 600 if ansible else 300
+        timeout = 900 if enabling_ansible else 300
```
Comment (mshriver) [inline cfme/utils/appliance/__init__.py]: So if I understand correctly, the UI role is enabled almost immediately, but the workers for the role are what take several minutes to start up.

If my understanding is correct, can't we leave this timeout to be 300 for the UI update, and apply the longer timeout to `self.wait_for_embedded_ansible` call on L1741

Comment (jkrocil) [reply cfme/utils/appliance/__init__.py]: Good catch! Yes, I agree with this, but Pete previously increased this to 600 and after talking to @quarckster I thought even 900 might be needed. But your are correct that 900 is really needed for `wait_for_embedded_ansible`, not for this. And `wait_for_embedded_ansible` is set to 900 by default so... I'll change this one back to 600, assuming Pete had a better reason to increase it to 600 than I have for the 900.

Comment (mshriver) [reply cfme/utils/appliance/__init__.py]: @psav can you confirm behavior here, UI doesn't actually need 600s to update roles, just the workers?  Ultimately if that's the case the timeout will never actually come into play anyway :)

**Comment by jkrocil** [comment]
None of the failures on PRT are related to this PR.

**Comment by jkrocil** [comment]
Tested locally; works.

**Comment by jkrocil** [comment]
2 approvals, appears to be working; merging to master.

**Pull request merged**